eva-server/src/test/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2IntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+})
+@ActiveProfiles(Profiles.TEST_MONGO_FACTORY)
+public class VariantWSServerV2IntegrationTest {
+
+    private static final String TEST_DB = "test-db";
+    @Rule
+    public MongoDbRule mongoDbRule = newMongoDbRule().defaultSpringMongoDb(TEST_DB);
+    @Autowired
+    MongoDbFactory mongoDbFactory;
+    @Autowired
+    private ApplicationContext applicationContext;
+    @Autowired
+    private TestRestTemplate restTemplate;
+    @Autowired
+    private VariantWithSamplesAndAnnotationsService service;
+
+    @Before
+    public void setUp() throws Exception { }
+
+    @Test
+    public void rootTestGetVariantsByVariantId() throws URISyntaxException {
+        String url = "/v2/variants/20:60100:A:T?species=mmusculus&assembly=grcm38";
+        List<VariantWithSamplesAndAnnotation> variantWithSamplesAndAnnotations = WSTestHelpers.testRestTemplateHelper(
+                url,restTemplate);
+        assertTrue(variantWithSamplesAndAnnotations.size() > 0);
+        assertTrue(variantWithSamplesAndAnnotations.get(0).getSourceEntries().size() == 0);
+        assertNull(variantWithSamplesAndAnnotations.get(0).getAnnotation());
+        assertTrue(variantWithSamplesAndAnnotations.get(0).getIds().size() > 0);
+    }
+
+    @Test
+    public void rootTestGetVariantsByNonExistingVariantId() throws URISyntaxException {
+        String url = "/v2/variants/10:0:A:T?species=mmusculus&assembly=grcm38";
+
+        ResponseEntity<QueryResponse<QueryResult<VariantWithSamplesAndAnnotation>>> response = restTemplate.exchange(
+                url, HttpMethod.GET, null,
+                new ParameterizedTypeReference<QueryResponse<QueryResult<VariantWithSamplesAndAnnotation>>>() {
+                });
+        assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+        assertEquals(0, response.getBody().getResponse().size());
+    }
+
+    @Test
+    public void rootTestForError() throws URISyntaxException {
+        String url = "/v2/variants/13:32889669:C:T?species=&assembly=grcm38";
+        assertEquals("Please specify a species", testForErrorHelper(url));
+    }
+
+    private String testForErrorHelper(String url) {
+        return WSTestHelpers.testRestTemplateHelperForError(url, restTemplate);
+    }
+
+    @Test
+    public void annotationEndPointTestExisting() throws URISyntaxException {
+        String url = "/v2/variants/20:60100:A:T/annotations?species=mmusculus&assembly=grcm38";
+        ResponseEntity<QueryResponse<QueryResult<Annotation>>> annotations = restTemplate.exchange(
+                url, HttpMethod.GET, null,
+                new ParameterizedTypeReference<QueryResponse<QueryResult<Annotation>>>() {
+                });
+        assertEquals(HttpStatus.OK, annotations.getStatusCode());
+        assertFalse(annotations.getBody().getResponse().get(0).getResult().get(0).getChromosome().isEmpty());
+    }
+
+    @Test
+    public void annotationEndPointTestNonExisting() throws URISyntaxException {
```

Comment:
Different situations of non-existence should be checked: one where the variant does not exist, and another where the variant exists but it is not annotated.

Reply:
I am a little bit confused regarding how to add this test since `/test-data/variants.json` file and both the documents in that have `annotation` and `source entries` fields present.
Is it okay to add extra data to this file to satisfy the requirement?

Reply:
Sure, it is totally valid according to our schema to have a variant without annotation. Variants should always have source entries though, the only part that could be missing are the statistics.

Reply:
Okay, I will add it :)